### DISCUSSION TITLE
Notifcation Improvements

### DISCUSSION THREAD

**Discussion by peaklabs-dev** [coollabsio/coolify#3163] [Improvement Requests]
## More notification options:
- Amazon SES for email
- slack
- whatsapp
- signal
- matrix
- teams
- push notifications for phone and pcs -> maybe browser based push notifications

-> and a webhook option ao we can use external services like zapier:
- webhook

## Other notification improvements:
- make it possible to add multiple notification providers of the same type (2 different smtp servers)
- backup SMTP server selection if 1 one fails
- add more notifications for more different things, backup, healthecks, proxy check, deployments, statistics (ram usage)...
- add more granular notification control for example select when to recive backup notifications -> only if it fails and only for a few production databases or add error notification -> when deployments fail and not always
....
- Send a notification always when an admin logs into coolify
- Update notifications for coolify panel updates
- Update notifications for docker images: https://github.com/coollabsio/coolify/issues/2500

**Comment by yohannlog** [comment]
Hi! I did a PR for the first request: #2601

**Comment by peaklabs-dev** [comment]
> Hi! I did a PR for the first request: #2594

Thank you so much.

**Comment by RayBB** [comment]
@yohannlog @ayntk-ai how would you feel about integrating with apprise so that just about any platform could be used to send notifications?
https://github.com/caronc/apprise 
changedetection.io uses this and it works quite nicely!

**Comment by peaklabs-dev** [comment]
> @yohannlog @ayntk-ai how would you feel about integrating with apprise so that just about any platform could be used to send notifications?
> https://github.com/caronc/apprise 
> changedetection.io uses this and it works quite nicely!

Looks nice, but I and also the maintainer I believe do not like external dependecies that are not absolutely necessary. So native integrations would be better. But overall I would have nothing against this, maybe in addition to native integrations for some more neach providers.

**Comment by peaklabs-dev** [comment]
I think native notifications should be (as they are the most common):
- [x] email
- [ ] Amazon SES for email
- [x] slack -> pull request open
- [x] discord
- [x] telegram
- [ ] whatsapp
- [ ] signal
- [ ] matrix
- [ ] teams
- [ ] push notifications for phone and pcs -> maybe browser based push notifications

-> and a webhook option ao we can use external services like zapier:
- [ ] webhook

**Comment by yohannlog** [comment]
> @yohannlog @ayntk-ai how would you feel about integrating with apprise so that just about any platform could be used to send notifications? https://github.com/caronc/apprise changedetection.io uses this and it works quite nicely!

Even if I like this solution, the problem is that you have to write the pages and code to support all the options for this package, so I don't know if it would be a time saver.

Besides, I don't know if this would be of much interest to the owner of this repo. Mainly for safety reasons.<

Edit: I will re-create the PR because I push it on the main (in my repo) and not a new branch

**Comment by RayBB** [comment]
For context, this is how change detection uses apprise.
As far as I know, it is only used on this page and the page for manual overrides.
<img width="1086" alt="image" src="https://github.com/coollabsio/coolify/assets/921217/eca20714-d074-44cc-8e12-e4f65c8202f7">

**Comment by yohannlog** [comment]
> For context, this is how change detection uses apprise. As far as I know, it is only used on this page and the page for manual overrides. <img alt="image" width="1086" src="https://private-user-images.githubusercontent.com/921217/341112230-eca20714-d074-44cc-8e12-e4f65c8202f7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTkwNjgxOTUsIm5iZiI6MTcxOTA2Nzg5NSwicGF0aCI6Ii85MjEyMTcvMzQxMTEyMjMwLWVjYTIwNzE0LWQwNzQtNDRjYy04ZTEyLWU0ZjY1YzgyMDJmNy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNjIyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDYyMlQxNDUxMzVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0xZWI2NDE5ZjkwODY0YmM2M2YzMTdlOGVjMjk1YjY3YWI0ZTJhMWJiNjU2ZDZjZDBjZjlmOTQ1NTFkNDE0ZThkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.6PvMF8K9nehpPW3Z_h9OlCSL9lTiaovT_yLZQZh23U4">

I will take a look. But the notification body is a great idea.

**Comment by madebyjulz** [comment]
It would be also great to just send notifications when anything doesnt works and not always.

**Comment by nullishamy** [comment]
@peaklabs-dev https://github.com/coollabsio/coolify/pull/3147 would appreciate some thoughts on this one - as it has a few upvotes now and would be greatly helpful for a deployment of coolify I plan on doing

**Comment by p3drosola** [comment]
I've been using the Discord notification option as a generic webhook notification. 

Instead of setting a discord url, I send the payload to n8n, and from there to slack or wherever. Maybe it would be enough to make the Discord notification a bit more generic, add support for a basic auth header or something and call it a day.

**Comment by ClusterDuckster** [comment]
I've taking a liking to [ntfy.sh](https://ntfy.sh/) for all my notifications. It has it's own [service template in coolify](https://github.com/coollabsio/coolify/blob/main/templates/compose/ntfy.yaml) alrerady, so it would feel natural to me to implement it as a notification target.

A webhook notification with an authentication option and ability to define a body template would be enough to enable basically all features of ntfy.sh.
A native integration example could be found at [uptime-kuma](https://github.com/louislam/uptime-kuma/blob/master/server/notification-providers/ntfy.js). That would also be nice for me, but I don't know how many people here use ntfy.sh. 

> I've been using the Discord notification option as a generic webhook notification.

I will try that with ntfy.sh as well, thanks for the tip!

**Comment by 50l3r** [comment]
Maybe it would be great to be able to configure notifications per project.

I have a discord server which I divide into channels according to each project. It would be great to be able to receive alerts about the deploys in each channel respectively 😁😁

**Comment by OliverGeneser** [comment]
@peaklabs-dev You mentioned wanting to keep external dependencies low. For the signal notifications, would the [Signal REST API](https://github.com/bbernhard/signal-cli-rest-api ) be acceptable, or would you prefer direct use of signal-cli?

**Reply by peaklabs-dev** [reply]
You can check how the current providers are implemented to get a good idea.
